subgraphs/euler-finance/src/mappings/handlers.ts

```diff
@@ -1,170 +1,316 @@
+
+  //for verification only
+  const eTokenAddress = Address.fromString(market.outputToken!);
+  const eToken = ERC20.bind(eTokenAddress);
+  const eTokenTotalSupply = eToken.totalSupply();
+  if (totalBalances.notEqual(eTokenTotalSupply)) {
+    log.warning("[logAssetStatus]market={}/{},totalBalances={},eTokenTotalSupply={}", [
+      market.name!,
+      market.id,
+      totalBalances.toString(),
+      eTokenTotalSupply.toString(),
+    ]);
+  }
+
+  // update prices, tvl, totalDepositBalanceUSD, totalBorrowBalanceUSD every 100 blocks (~20 min)
+  if (event.block.number.ge(protocol._lastUpdateBlockNumber!.plus(BIGINT_HUNDRED))) {
+    const eulerContract = Euler.bind(Address.fromString(EULER_ADDRESS));
+    const execProxyAddress = eulerContract.moduleIdToProxy(MODULEID__EXEC);
+    let totalDepositBalanceUSD = BIGDECIMAL_ZERO;
+    let totalBorrowBalanceUSD = BIGDECIMAL_ZERO;
+    for (let i = 0; i < protocol._marketIDs!.length; i++) {
+      const mrktID = protocol._marketIDs![i];
+      const mrkt = getOrCreateMarket(mrktID);
+      const tkn = getOrCreateToken(Address.fromString(mrkt.inputToken));
+      const currPriceUSD = updatePrices(execProxyAddress, mrkt, event);
+      const underlyingPriceUSD = currPriceUSD ? currPriceUSD : mrkt.inputTokenPriceUSD;
+      // mark-to-market
+      mrkt.totalDepositBalanceUSD = bigIntToBDUseDecimals(mrkt.inputTokenBalance, tkn.decimals).times(
+        underlyingPriceUSD,
+      );
+      mrkt.totalBorrowBalanceUSD = bigIntToBDUseDecimals(mrkt._totalBorrowBalance!, tkn.decimals).times(
+        underlyingPriceUSD,
+      );
+      mrkt.totalValueLockedUSD = mrkt.totalDepositBalanceUSD;
+      mrkt.save();
+
+      log.debug(
+        "[handleAssetStatus]market={}/{},block={},totalBorrowBalanceUSD={},totalDepositBalanceUSD={},exchangeRate={},PriceUSD={}",
+        [
+          mrkt.name!,
+          mrkt.id,
+          event.block.number.toString(),
+          mrkt.totalBorrowBalanceUSD.toString(),
+          mrkt.totalDepositBalanceUSD.toString(),
+          mrkt.exchangeRate!.toString(),
+          underlyingPriceUSD.toString(),
+        ],
+      );
+
+      totalDepositBalanceUSD = totalDepositBalanceUSD.plus(mrkt.totalDepositBalanceUSD);
+      totalBorrowBalanceUSD = totalBorrowBalanceUSD.plus(mrkt.totalBorrowBalanceUSD);
+    }
+
+    protocol.totalDepositBalanceUSD = totalDepositBalanceUSD;
+    protocol.totalBorrowBalanceUSD = totalBorrowBalanceUSD;
+    protocol.totalValueLockedUSD = totalDepositBalanceUSD;
+    protocol._lastUpdateBlockNumber = event.block.number;
+    protocol.save();
+  }
+
+  log.debug("[handleAssetStatus]block={},protocol.totalBorrowBalanceUSD={},protocol.totalDepositBalanceUSD={}", [
+    event.block.number.toString(),
+    protocol.totalBorrowBalanceUSD.toString(),
+    protocol.totalDepositBalanceUSD.toString(),
+  ]);
+
+  snapshotFinancials(event.block, BIGDECIMAL_ZERO, null, protocol);
+
+  assetStatus.totalBorrows = totalBorrows;
```

Comment:
I'm going to argue that this would increase readability if you put this after you grab those values from the event. @jaimehgb do you have any thoughts here?

Reply:
Hmm, I agree in a way because the function is very big, in my opinion, and after reading through it all, when you reach the end, you might not have clear where the `totalBorrows` value came from. 

But if we grabbed the value from the event, we should do it too for `totalBalances`, `reserveBalance`, `underlying`, `poolSize`, ... to keep it consistent. It might become too verbose, so I guess it's matter of preference.

What I'd definitely do is breaking down the function into smaller ones, reducing its size so that it doesn't matter if you use `totalBorrows` or `event.params.totalBorrows` because either way will be clear enough.

Reply:
Moved up the chunk & put some of the code into functions.